File: Atoms/com.AndrewHazelden.UIManagerLuaExamples/Scripts/Comp/UI Manager/CheckBox.py
ui = fu.UIManager
disp = bmd.UIDispatcher(ui)

dlg = disp.AddWindow({ "WindowTitle": "My First Window", "ID": "MyWin", "Geometry": [ 100, 100, 400, 200 ], },
	[
		ui.VGroup({ "Spacing": 0, },
		[
			# Add your GUI elements here:
			ui.CheckBox({ "ID": "MyCheckbox", "Text": "The Checkbox Label"}),
		]),
	])

itm = dlg.GetItems()

# The window was closed
def _func(ev):
	disp.ExitLoop()
dlg.On.MyWin.Close = _func

# Add your GUI element based event functions here:

def _func(ev):
	print( "[Checkbox] " + str(itm['MyCheckbox'].Checked))
dlg.On.MyCheckbox.Clicked = _func


dlg.Show()
disp.RunLoop()
dlg.Hide()

File: Atoms/com.AndrewHazelden.UIManagerLuaExamples/Scripts/Comp/UI Manager/CheckBox.lua
local ui = fu.UIManager
local disp = bmd.UIDispatcher(ui)
local width,height = 400,200

win = disp:AddWindow({
	ID = 'MyWin',
	WindowTitle = 'My First Window',
	Geometry = {100, 100, width, height},
	Spacing = 10,

	ui:VGroup{
		ID = 'root',
		Margin = 0,

		-- Add your GUI elements here:
		ui:CheckBox{
			ID = 'MyCheckbox',
			Text = 'The Checkbox Label',
		},
	},
})

-- The window was closed
function win.On.MyWin.Close(ev)
	disp:ExitLoop()
end

-- Add your GUI element based event functions here:
itm = win:GetItems()

function win.On.MyCheckbox.Clicked(ev)
	print('[Checkbox] ' .. tostring(itm.MyCheckbox.Checked))
end

win:Show()
disp:RunLoop()
win:Hide()
